transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -468,6 +496,11 @@ static jint netty_epoll_native_recvmmsg0(JNIEnv* env, jclass clazz, jint fd, jbo
 
         msg[i].msg_hdr.msg_name = addr + i;
         msg[i].msg_hdr.msg_namelen = (socklen_t) addrSize;
+
+        if (cntrlbuf != NULL) {
+            msg[i].msg_hdr.msg_control =  cntrlbuf + len * storageSize;
+            msg[i].msg_hdr.msg_controllen = storageSize;
```

Comment:
Why is this `len * storageSize`? Shouldn't it be `i * storageSize`?

Reply:
No the Len should be big enough to contain the address .

Reply:
But `cntrlbuf` is a pointer to a buffer of size `storageSize * len`.
So `cntrlbuf + len * storageSize` is a pointer to just past the end of `cntrlbuf` (the multiplication is evaluated before the addition).
And you assign this value to `msg[i].msg_hdr.msg_control` for every `i`.